### PR TITLE
Add doc for the latest "MacCatalyst is superset of iOS" updates

### PR DESCRIPTION
**Pull request by buyaa-n**
## Summary

Recently we updated the Platform Compatibility Analyzer to support a relationship of "MacCatalyst is a superset of IOS" using SupportedOSPlatformGuard attributes on the OperatingSystem method. We need to update the doc accordingly 

Related PR: https://github.com/dotnet/roslyn-analyzers/pull/5266

```csharp
  // Macatalyst is a superset of iOS therefore supported on iOS and MacCatalyst  
  [SupportedOSPlatform("iOS")]
  public void ApiOnlySupportedOnIOSAndMacCatalyst() { }

  // Does not imply iOS, only supported on MacCatalyst
  [SupportedOSPlatform("MacCatalyst")]
  public void ApiOnlySupportedOnMacCatalyst() { }

  [SupportedOSPlatform("iOS")] // Supported on iOS and MacCatalyst  
  [UnsupportedOSPlatform("MacCatalyst")] // Removes implied MacCatalyst support
  public void ApiOnlySupportedOnIos() { }

  // Unsupported on iOS and MacCatalyst  
  [UnsupportedOSPlatform("iOS")]
  public void ApiUnsupportedOnIOSAndMacCatalyst();

  // Does not imply iOS, only unsupported on MacCatalyst
  [UnsupportedOSPlatform("MacCatalyst")]
  public void ApiUnsupportedOnMacCatalyst() { }

  [UnsupportedOSPlatform("iOS")] // Unsupported on iOS and MacCatalyst  
  [SupportedOSPlatform("MacCatalyst")] // Removes implied MacCatalyst unsupport
  public void ApiUnsupportedOnIos() { }
```

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca1416.md [hunk 1]
```diff
@@ -119,22 +119,22 @@ The recommended way to deal with violations is to make sure you only call platfo
...
+  [SupportedOSPlatform("ios14.0")] // MacCatalyst is superset of iOS, therefore it also supported on MacCatalyst 14.0
+  public void SupportedOnWindowsIos14AndMacCatalyst14() { }
 
   public void Caller()
   {
       LinuxOnlyApi(); // This call site is reachable on all platforms. 'LinuxOnlyApi()' is only supported on: 'linux'.
 
-      SupportedOnWindowsAndIos14(); // This call site is reachable on all platforms. 'SupportedOnWindowsAndIos14()'
-                                    // is only supported on: 'windows', 'ios' 14.0 and later
+      SupportedOnWindowsIos14AndMacCatalyst14(); // This call site is reachable on all platforms. 'SupportedOnWindowsAndIos14()'
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca1416.md:131]:
```suggestion
      SupportedOnWindowsIos14AndMacCatalyst14(); // This call site is reachable on all platforms. 'SupportedOnWindowsIos14AndMacCatalyst14()'
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 2]
```diff
@@ -23,16 +23,20 @@ The platform compatibility analyzer is one of the Roslyn code quality analyzers.
 ## How the analyzer determines platform dependency
 
 - An **unattributed API** is considered to work on **all OS platforms**.
-- An API marked with `[SupportedOSPlatform("platform")]` is considered only portable to the specified OS `platform`.
-  - The attribute can be applied multiple times to indicate **multiple platform support** (`[SupportedOSPlatform("windows"), SupportedOSPlatform("Android29.0")]`).
+- An API marked with `[SupportedOSPlatform("platform")]` is considered only portable to the specified and parent OS `platform(s)`.
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:26]:
```suggestion
- An API marked with `[SupportedOSPlatform("platform")]` is considered only portable to the specified platform and its parent OS.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 3]
```diff
@@ -23,16 +23,20 @@ The platform compatibility analyzer is one of the Roslyn code quality analyzers.
...
+  - The attribute can be applied multiple times to indicate **multiple platform support**, for example `[SupportedOSPlatform("windows"), SupportedOSPlatform("Android29.0")]`.
+  - For [**related** platforms](#related-platforms) the attribute also imply its parent support:
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:28]:
```suggestion
  - For [**related** platforms](#related-platforms), the attribute implies that the parent is also supported.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 4]
```diff
@@ -23,16 +23,20 @@ The platform compatibility analyzer is one of the Roslyn code quality analyzers.
...
+    - For example `[SupportedOSPlatform("iOS")` annotation imply that the API is supported on `iOS` and also supported on its parent platform: `MacCatalyst`.
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:29]:
```suggestion
    - For example, `[SupportedOSPlatform("iOS")]` implies that the API is supported on `iOS` and also on its parent platform, `MacCatalyst`.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 5]
```diff
@@ -23,16 +23,20 @@ The platform compatibility analyzer is one of the Roslyn code quality analyzers.
...
     - **Warns** if the project is cross-platform and calls platform-specific APIs (for example, a Windows-specific API called from cross platform TFM `<TargetFramework>net5.0</TargetFramework>`).
     - **Doesn't warn** if the platform-specific API is referenced within a project that targets any of the **specified platforms** (for example, for a Windows-specific API called from a project targeting windows `<TargetFramework>net5.0-windows</TargetFramework>`).
     - **Doesn't warn** if the platform-specific API call is guarded by corresponding platform-check methods (for example, a Windows-specific API call guarded by `OperatingSystem.IsWindows()`).
     - **Doesn't warn** if the platform-specific API is referenced from the same platform-specific context (**call site also attributed** with `[SupportedOSPlatform("platform")`).
-- An API marked with `[UnsupportedOSPlatform("platform")]` is considered unsupported only on the specified OS `platform` but supported for all other platforms.
+- An API marked with `[UnsupportedOSPlatform("platform")]` is considered unsupported on the specified and parent OS `platform(s)` but supported for all other platforms.
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:36]:
```suggestion
- An API marked with `[UnsupportedOSPlatform("platform")]` is considered to be unsupported on the specified platform and its parent OS, but supported for all other platforms.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 6]
```diff
@@ -23,16 +23,20 @@ The platform compatibility analyzer is one of the Roslyn code quality analyzers.
...
   - The attribute can be applied multiple times with different platforms, for example, `[UnsupportedOSPlatform("iOS"), UnsupportedOSPlatform("Android29.0")]`.
+  - For [**related** platforms](#related-platforms) the attribute also imply its parent unsupport:
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:38]:
```suggestion
  - For [**related** platforms](#related-platforms), the attribute implies that the parent platform is also unsupported.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 7]
```diff
@@ -23,16 +23,20 @@ The platform compatibility analyzer is one of the Roslyn code quality analyzers.
...
+    - For example: `[UnsupportedOSPlatform("iOS")` annotation imply that the API is unsupported on `iOS` and also unsupported on its parent platform: `MacCatalyst`.
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:39]:
```suggestion
    - For example, `[UnsupportedOSPlatform("iOS")]` implies that the API is unsupported on `iOS` and also on its parent platform, `MacCatalyst`.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 8]
```diff
@@ -56,7 +60,45 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+  // Unsupported on iOS and MacCatalyst  
+  [UnsupportedOSPlatform("iOS")]
+  public void ApiUnsupportedOnIOSAndMacCatalyst();
+
+  // Does not imply iOS, only unsupported on MacCatalyst
+  [UnsupportedOSPlatform("MacCatalyst")]
+  public void ApiUnsupportedOnMacCatalyst() { }
+
+  [UnsupportedOSPlatform("iOS")] // Unsupported on iOS and MacCatalyst  
+  [SupportedOSPlatform("MacCatalyst")] // Removes implied MacCatalyst unsupport
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:97]:
```suggestion
  [SupportedOSPlatform("MacCatalyst")] // Removes implied MacCatalyst unsupportedness
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 9]
```diff
@@ -56,7 +60,45 @@ The analyzer does not check target framework moniker (TFM) target platforms from
   > [!NOTE]
   > If the *AssemblyInfo.cs* file generation is disabled for the project (that is, the `<GenerateAssemblyInfo>` property is set to `false`), the required assembly level `SupportedOSPlatform` attribute can't be added by MSBuild. In this case, you could see warnings for a platform-specific APIs usage even if you're targeting that platform. To resolve the warnings, enable the *AssemblyInfo.cs* file generation or add the attribute manually in your project.
 
-### Advanced scenarios for combining attributes
+### Related platforms
+
+In .NET 6.0 we added new scenario for platforms relation where a platform could be a superset of another platform and the child platform annotation would imply the parent platform support/unsupport. If any platform check method of <xref:System.OperatingSystem> type has a `SupportedOSPlatformGuard("parentPlatform")]` attribute then that `parentPlatform` is accounted as a superset of the OS platform the method is checking.
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:65]:
```suggestion
.NET 6 introduces the concept of *related platforms*, where one platform can be a superset of another platform. An annotation for the child platform implies the same support (or lack thereof) for the parent platform. If a platform check method in the <xref:System.OperatingSystem> type has a `SupportedOSPlatformGuard("parentPlatform")]` attribute, then `parentPlatform` is considered a superset of the OS platform the method is checking.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 10]
```diff
@@ -56,7 +60,45 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+
+For example <xref:System.OperatingSystem.IsIOS?displayProperty=nameWithType> has `SupportedOSPlatformGuard("MacCatalyst")]` therefore:
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:67]:
```suggestion
For example, the <xref:System.OperatingSystem.IsIOS?displayProperty=nameWithType> method is attributed `[SupportedOSPlatformGuard("MacCatalyst")]`. Therefore, the following statements apply:
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 11]
```diff
@@ -56,7 +60,45 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+
+  1. <xref:System.OperatingSystem.IsIOS?displayProperty=nameWithType> and <xref:System.OperatingSystem.IsIOSVersionAtLeast%2A?displayProperty=nameWithType> methods not only checking `iOS` platform, it also can be used for checking the platform `MacCatalyst`.
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:69]:
```suggestion
  - The <xref:System.OperatingSystem.IsIOS?displayProperty=nameWithType> and <xref:System.OperatingSystem.IsIOSVersionAtLeast%2A?displayProperty=nameWithType> methods check not only the `iOS` platform, but also the `MacCatalyst` platform.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 12]
```diff
@@ -56,7 +60,45 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+  2. `[SupportedOSPlatform("iOS")` annotation imply that the API is supported on `iOS` and also supported on its parent platform: `MacCatalyst`.
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:70]:
```suggestion
  - `[SupportedOSPlatform("iOS")]` implies that the API is supported on `iOS` and also on its parent platform, `MacCatalyst`.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 13]
```diff
@@ -56,7 +60,45 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+    - One can use `[UnsSupportedOSPlatform("MacCatalyst")]` attribute to exclude this implied support.
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:71]:
```suggestion
    You can use the `[UnsSupportedOSPlatform("MacCatalyst")]` attribute to exclude this implied support.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 14]
```diff
@@ -56,7 +60,45 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+  3. `[UnsupportedOSPlatform("iOS")` implies that the API is not supported on `iOS` and `Maccatalyst`
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:72]:
```suggestion
  3. `[UnsupportedOSPlatform("iOS")` implies that the API is not supported on `iOS` and `Maccatalyst`.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 15]
```diff
@@ -56,7 +60,45 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+    - One can use `[SupportedOSPlatform("MacCatalyst")]` attribute to exclude this implied unsupport.
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:73]:
```suggestion
    - You can use the `[SupportedOSPlatform("MacCatalyst")]` attribute to exclude this implied lack of support.
```

**Review by gewarren — COMMENTED** [review]
It might also be good to include a table showing the support combinations. I'll see if I can come up with one.

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 16]
```diff
@@ -56,7 +60,45 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+
```
Comment (gewarren) [inline docs/standard/analyzers/platform-compat-analyzer.md:74]:
```suggestion

Consider the following support matrix.

| | `SupportedOSPlatform(child)` | `SupportedOSPlatform(parent)` | `UnsupportedOSPlatform(child)` | `UnsupportedOSPlatform(parent)` |
| - | - | - | - | - |
| **Child OS** | ✔️ | ❌ | ❌ | ✔️ |
| **Parent OS** | ✔️ | ✔️ | ❌ | ❌ |

The following code snippet shows how you can combine attributes to set the right level of support.

```

**Comment by buyaa-n** [comment]
> It might also be good to include a table showing the support combinations. I'll see if I can come up with one.

Thank you, i have updated the table bit and applied with all other suggestions

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 17]
```diff
@@ -56,7 +60,54 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
 
-### Advanced scenarios for combining attributes
+### Related platforms
+
+.NET 6 introduces the concept of *related platforms*, where one platform can be a superset of another platform. An annotation for the child platform implies the same support (or lack thereof) for the parent platform. If a platform check method in the <xref:System.OperatingSystem> type has a `SupportedOSPlatformGuard("parentPlatform")]` attribute, then `parentPlatform` is considered a superset of the OS platform the method is checking.
+
+For example, the <xref:System.OperatingSystem.IsIOS?displayProperty=nameWithType> method is attributed `[SupportedOSPlatformGuard("MacCatalyst")]`. Therefore, the following statements apply:
+
+  - The <xref:System.OperatingSystem.IsIOS?displayProperty=nameWithType> and <xref:System.OperatingSystem.IsIOSVersionAtLeast%2A?displayProperty=nameWithType> methods check not only the `iOS` platform, but also the `MacCatalyst` platform.
```
Comment (buyaa-n) [inline docs/standard/analyzers/platform-compat-analyzer.md:69]:
```suggestion
- The <xref:System.OperatingSystem.IsIOS?displayProperty=nameWithType> and <xref:System.OperatingSystem.IsIOSVersionAtLeast%2A?displayProperty=nameWithType> methods check not only the `iOS` platform, but also the `MacCatalyst` platform.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 18]
```diff
@@ -56,7 +60,54 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+  - `[SupportedOSPlatform("iOS")]` implies that the API is supported on `iOS` and also on its parent platform, `MacCatalyst`.
```
Comment (buyaa-n) [inline docs/standard/analyzers/platform-compat-analyzer.md:70]:
```suggestion
- `[SupportedOSPlatform("iOS")]` implies that the API is supported on `iOS` and also on its parent platform, `MacCatalyst`.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 19]
```diff
@@ -56,7 +60,54 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+    You can use the `[UnsSupportedOSPlatform("MacCatalyst")]` attribute to exclude this implied support.
```
Comment (buyaa-n) [inline docs/standard/analyzers/platform-compat-analyzer.md:71]:
```suggestion
  - You can use the `[UnsSupportedOSPlatform("MacCatalyst")]` attribute to exclude this implied support.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 20]
```diff
@@ -56,7 +60,54 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+  3. `[UnsupportedOSPlatform("iOS")` implies that the API is not supported on `iOS` and `Maccatalyst`.
```
Comment (buyaa-n) [inline docs/standard/analyzers/platform-compat-analyzer.md:72]:
```suggestion
- `[UnsupportedOSPlatform("iOS")` implies that the API is not supported on `iOS` and `Maccatalyst`.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 21]
```diff
@@ -56,7 +60,54 @@ The analyzer does not check target framework moniker (TFM) target platforms from
...
+    - You can use the `[SupportedOSPlatform("MacCatalyst")]` attribute to exclude this implied lack of support.
```
Comment (buyaa-n) [inline docs/standard/analyzers/platform-compat-analyzer.md:73]:
```suggestion
  - You can use the `[SupportedOSPlatform("MacCatalyst")]` attribute to exclude this implied lack of support.
```

File: docs/standard/analyzers/platform-compat-analyzer.md [hunk 22]
```diff
@@ -82,7 +133,7 @@ The analyzer does not check target framework moniker (TFM) target platforms from
     ```
 
   - **Inconsistent list**. If the lowest version for some platforms is `[SupportedOSPlatform]` while it is `[UnsupportedOSPlatform]` for other platforms, it's considered to be inconsistent, which is not supported for the analyzer. If inconsistency occurs, the analyzer ignores the `[UnsupportedOSPlatform]` platforms.
-  - If the lowest versions of `[SupportedOSPlatform]` and `[UnsupportedOSPlatform]` attributes are equal, the analyzer considers the platform as part of the **Supported only list**.
+    - If the lowest versions of `[SupportedOSPlatform]` and `[UnsupportedOSPlatform]` attributes are equal, the analyzer considers the platform as part of the **Supported only list**.
```
Comment (jeffhandley) [inline docs/standard/analyzers/platform-compat-analyzer.md:136]: Is this spacing correct?

**Review by jeffhandley — APPROVED** [review]
I left one comment where I wasn't sure if the spacing/indentation was correct.  Beyond that, the only feedback I have is that we're using a few different terms:

1. Related
2. Superset
3. Parent/Child

If we're going to use parent/child, then "related" is a good word that refers to the relationship between parent/child. But I don't know that parent/child is the most intuitive or helpful term for the concept; I think superset/subset is more informative of the direction of the relationship.

If we had superset/subset instead of parent/child, then I'm not sure that "related" is best, but I also can't think of a good replacement.

File: docs/standard/analyzers/platform-compat-analyzer.md (hunk 22 above)
Comment (buyaa-n) [reply docs/standard/analyzers/platform-compat-analyzer.md:136]: Yes, this is a sub scenario of "Inconsistent list"

**Comment by buyaa-n** [comment]
> If we're going to use parent/child, then "related" is a good word that refers to the relationship between parent/child. But I don't know that parent/child is the most intuitive or helpful term for the concept; I think superset/subset is more informative of the direction of the relationship.
> 
> If we had superset/subset instead of parent/child, then I'm not sure that "related" is best, but I also can't think of a good replacement.

Right, also i noticed that we used `parent/child` for nested attributes, like `Assembly` level attributes are parent for `Type/API` level attributes, etc. Therefore i updated the `parent/child` with `superset/subset` and instead `related attributes` used `attributes relation`. Please take another look

**Comment by gewarren** [comment]
I left some suggestions in a PR since that was easier than line by line suggestions. https://github.com/buyaa-n/docs/pull/2

**Comment by buyaa-n** [comment]
> I left some suggestions in a PR since that was easier than line by line suggestions. [buyaa-n#2](https://github.com/buyaa-n/docs/pull/2)

Thank you @gewarren that is great, sorry for the late reply, the notification was lost within my emails, I just merged your PR.

**Review by gewarren — APPROVED** [review]

**Pull request merged**